Feature/bhoeve

benjamin-hoeve opened the pull request.

Commit:
Several fixes

- Changed Thread loop to Task
- JIngle volume control based on percentage. The default percentage can be set in the appsettings
- Removed unnecessary packages from youtube musicservice. Removed Google Api, we only use Youtube Explode now
- WIP => show errormessage from musicservice in frond-end

Commit:
Removed YouTube ApiKey, no longer necessary

Commit:
Merged develop into feature/bhoeve

Commit:
Refactor startup class, moved init of musicservices to a method

Commit:
Error messages in front-end. When an error occurs, it'll be shown in the front-end

Commit:
Merge branch 'develop' of https://github.com/Bloemert/musicus into feature/bhoeve

b-hoeve committed:
Next track is now based on events. When a song ends, an event is triggered.

b-hoeve committed:
Merge branch 'develop' of https://github.com/Bloemert/musicus into feature/bhoeve

Commit:
Add MusicServices in Startup configuration.

- Add Spotify & YouTube by adding services.AddSpotifyMusicService() or AddYouTubeMusicService() to the Startup class

Commit:
Auto play next song bugfix
Event was triggered multiple times

Commit:
Init volume at startup.
The musicservices are configured with a default volume at startup. This is to prevent volume differences between MusicServices

Commit:
Cleanup

Commit:
When a song couldn't be played, an error message is send to all clients to notify them.
If there is another song in queue, that song is played

lt-gerjan commented on src/Musicus.Abstractions/Services/IMusicService.cs:
```diff
@@ -4,6 +4,8 @@
 
 namespace Musicus.Abstractions.Services
 {
+	public delegate void TrackEndHandler();
```
Why not an `Action`?

lt-gerjan commented on src/Musicus.Abstractions/Models/IActionResult.cs:
```diff
@@ -0,0 +1,9 @@
+{
+	public interface IActionResult<T>
+	{
+		bool Succeed { get; set; }
```
Succeed? = Slagen?

lt-gerjan commented on src/Musicus.MusicServices/Musicus.SpotifyService/SpotifyServiceConfiguration.cs:
```diff
@@ -0,0 +1,23 @@
+{
+	public static class SpotifyServiceConfiguration
+	{
+		public static IServiceCollection AddSpotifyMusicService(this IServiceCollection services, IConfiguration config)
```
Cool, nice idea :)

lt-gerjan commented on src/Musicus.MusicServices/Musicus.YouTubeService/YouTubeMusicService.cs:
```diff
@@ -8,27 +8,44 @@ namespace YouTubeService
 
-		public async Task<IMusicServiceStatus> GetStatusAsync() => await Task.Run(() => YouTubeHelper.GetStatus());
+		public async Task<IActionResult<IMusicServiceStatus>> GetStatusAsync()
+			 => await Task.Run(() => YouTubeHelper.GetStatus());
```
Not liking all the Task.Run wrappers

lt-gerjan reviewed (approved):
Approved when addressing feedback

b-hoeve committed:
Refactoring
- Removed Task wrappers
- An exception is thrown when no spotify credentials are given
- Changed delegate to action
